webapp/portlet/src/main/webapp/WEB-INF/jsp/portlet/bannerLogo.jsp

```diff
@@ -51,7 +56,21 @@
 
     portalPath = portalConfigService.getUserHomePage(request.getRemoteUser());
     if (portalPath == null) {
-      portalPath = defaultHomePath;
+      SiteFilter siteFilter = new SiteFilter();
+      siteFilter.setExcludedSiteName(portalConfigService.getGlobalPortal());
+      siteFilter.setSiteType(SiteType.PORTAL);
+      siteFilter.setSortByDisplayOrder(true);
+      siteFilter.setFilterByDisplayed(true);
+      siteFilter.setDisplayed(true);
+      List<PortalConfig> portalConfigList = portalConfigService.getSites(siteFilter);
+      if (portalConfigList != null && !portalConfigList.isEmpty()) {
+        String defaultPortal = portalConfigList.get(0).getName();
+        Collection<UserNode> userNodes = portalConfigService.getSiteNavigations(defaultPortal, authenticatedUser, requestContext.getRequest());
+        UserNode userNode = portalConfigService.getFirstAllowedPageNode(userNodes);
+        portalPath = portalConfigService.getDefaultUri(userNode, defaultPortal);
+      } else {
+        portalPath = defaultHomePath;
+      }
```

Comment:
Shouldn't we add this logic to compute the default `portalPath` from Service layer ?

Reply:
fixed